polaris/polaris/management/commands/process_pending_deposits.py

```diff
@@ -729,14 +866,185 @@ async def save_as_pending_signatures(cls, transaction, server):
             transaction.envelope_xdr = envelope.to_xdr()
             transaction.pending_signatures = True
             transaction.status = Transaction.STATUS.pending_anchor
-        transaction.pending_execution_attempt = False
         await sync_to_async(transaction.save)()
         await maybe_make_callback_async(transaction)
 
     @classmethod
     def handle_error(cls, transaction, message):
         transaction.status_message = message
         transaction.status = Transaction.STATUS.error
-        transaction.pending_execution_attempt = False
         transaction.save()
         logger.error(message)
+
+    @classmethod
+    def handle_transaction_submission_exception(cls, transaction, exception, message):
+        if type(exception) is TransactionSubmissionBlocked:
+            transaction.submission_status = Transaction.SUBMISSION_STATUS.blocked
+        elif type(exception) is TransactionSubmissionFailed:
+            transaction.status = Transaction.STATUS.error
+            transaction.submission_status = Transaction.SUBMISSION_STATUS.failed
+        elif type(exception) is TransactionSubmissionPending:
+            transaction.submission_status = Transaction.SUBMISSION_STATUS.pending
+        transaction.status_message = message
+        transaction.save()
+
+    @classmethod
+    def update_heartbeat(cls, key):
+        with django.db.transaction.atomic():
+            heartbeat = PolarisHeartbeat.objects.filter(key=key)[0]
+            heartbeat.last_heartbeat = datetime.datetime.now(datetime.timezone.utc)
+            heartbeat.save()
+        return
```

Comment:
You can just use an update query here, vs. a get + update. That way you don't need an atomic block.

```python
PolarisHeartbeat.objects.update(last_heartbeat=datetime.datetime.now(datetime.timezone.utc))
```